toolchain/lexer/tokenized_buffer.cpp

```diff
@@ -716,24 +830,16 @@ auto TokenizedBuffer::Lex(SourceBuffer& source, DiagnosticConsumer& consumer)
   ErrorTrackingDiagnosticConsumer error_tracking_consumer(consumer);
   Lexer lexer(buffer, error_tracking_consumer);
 
+  // Build a table of function pointers that we can use to dispatch to the
+  // correct lexer routine based on the first byte of source text.
+  constexpr Lexer::DispatchTableT DispatchTable = Lexer::MakeDispatchTable();
+
   llvm::StringRef source_text = source.text();
   while (lexer.SkipWhitespace(source_text)) {
```

Comment:
Should we do whitespace skipping as part of the dispatch loop too? It seems unnecessary to first check whether we're at a whitespace character and then check again what kind of token we're forming, in the case where there isn't whitespace between tokens.